### PR TITLE
feat: Attach the user's name to the user message payload if known

### PR DESCRIPTION
**Pull request by Teagan42**
# Motivation

I wanted my assistant to know who it was talking to.

# Approach

If there is a user_id attached to the context that is provided via `user_input`, retrieve the name of the user from HASS's authentication manager. This primarily applies when interacting with the UI or invoking the process service from a user context.

# Future Work

## Wake Words
Add voice recognition to wake word detection to differentiate the user who started the conversation

## Full Feature
Add voice recognition to differentiate the user who is currently speaking - allowing for multiple participate in a conversation

### CODE REVIEW

**Comment by jekalmin** [comment]
Thank you for your work!
I just tried and seems to work well for me.

One concern I have is that since this is the first custom field(not required for openai) to put in the message, wonder if everyone want it to be added or not.

**Comment by Teagan42** [comment]
I was thinking about adding it to the core project as well

**Comment by jekalmin** [comment]
If you don't mind, please let me keep this PR open for a while.
I am quite not sure whether this should be configurable or mandatory yet.

**Comment by Teagan42** [comment]
No problem

**Comment by Teagan42** [comment]
Went ahead and added it to the `config_flow` as you’re right, it should be optional for users.

**Comment by jekalmin** [comment]
Thanks for going ahead!
I will try this today or tomorrow, and let you know if anything is needed.

**Review by jekalmin — CHANGES_REQUESTED** [review]
Could you add a label of new field in translations?
- [en.json](https://github.com/jekalmin/extended_openai_conversation/blob/13867d176c7740c9b55678b22c6cfb54fd4d9f1e/custom_components/extended_openai_conversation/translations/en.json)
- [ko.json](https://github.com/jekalmin/extended_openai_conversation/blob/13867d176c7740c9b55678b22c6cfb54fd4d9f1e/custom_components/extended_openai_conversation/translations/ko.json)
- [strings.json](https://github.com/jekalmin/extended_openai_conversation/blob/13867d176c7740c9b55678b22c6cfb54fd4d9f1e/custom_components/extended_openai_conversation/strings.json)

File: custom_components/extended_openai_conversation/config_flow.py [hunk 1]
```diff
@@ -194,4 +198,11 @@ def openai_config_option_schema(options: MappingProxyType[str, Any]) -> dict:
             description={"suggested_value": options.get(CONF_FUNCTIONS)},
             default=DEFAULT_CONF_FUNCTIONS_STR,
         ): TemplateSelector(),
+        vol.Optional(
+            CONF_ATTACH_USERNAME,
+            description={
+                "suggested_value": options[CONF_ATTACH_USERNAME]
```
[the remark this answers is not in the document]
Comment (Teagan42) [reply custom_components/extended_openai_conversation/config_flow.py:204]: Good catch

File: README.md [hunk 2]
```diff
@@ -55,6 +56,7 @@ https://github.com/jekalmin/extended_openai_conversation/assets/2917984/64ba656e
 By clicking a button from Edit Assist, Options can be customized.<br/>
 Options include [OpenAI Conversation](https://www.home-assistant.io/integrations/openai_conversation/) options and two new options. 
 
+- `Attach Username`: Pass the active user's name (if applicaple) to OpenAI via rhe message payload. Currently, this only applies to conversations through the UI or REST API.
```
Comment (jekalmin) [inline README.md:59]: Just curious of what `rhe` stands for.

**Review by jekalmin — COMMENTED** [review]
Thanks for a change!
I left a quick comment below :D

File: README.md (hunk 2 above)
Comment (Teagan42) [reply README.md:59]: Typeo

Comment (Teagan42) [reply README.md]: Resolved

**Review by jekalmin — APPROVED** [review]
Thanks for your work!
I will apply this change in next release (currently released in [0.0.9-beta](https://github.com/jekalmin/extended_openai_conversation/releases/tag/0.0.9-beta1))

**Pull request merged**